test/metrics/metrics_test.go

```diff
@@ -0,0 +1,735 @@
+
+				t.Run("verify metrics", func(t *testing.T) {
+					metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 0)         // unchanged: user signup already existed
+					metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 1) // user approved
+					metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 0)   // unchanged: banneduser already existed
+					metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 1, "domain", "external")
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 1, "cluster_name", memberAwait.ClusterName)  // user provisioned on member1
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+				})
+			})
+		})
+
+		t.Run("manually created usersignup with preexisting banneduser", func(t *testing.T) {
+			// Get metrics assertion helper for testing metrics
+			metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+
+			hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().Enabled())
+
+			id := uuid.Must(uuid.NewV4()).String()
+			email := "testuser" + id + "@test.com"
+			banUser(t, hostAwait, email)
+
+			// For this test, we don't want to create the UserSignup via the registration service
+			// Instead, we want to confirm the behaviour when a UserSignup with a banned email address is created manually
+			userSignup := NewUserSignup(t, hostAwait, "testuser"+id, email)
+			userSignup.Spec.TargetCluster = memberAwait.ClusterName
+
+			// Create the UserSignup via the Kubernetes API
+			err := hostAwait.CreateWithCleanup(context.TODO(), userSignup)
+			require.NoError(t, err)
+			t.Logf("user signup '%s' created", userSignup.Name)
+
+			// Check the UserSignup is created
+			userSignup, err = hostAwait.WaitForUserSignup(userSignup.Name)
+			require.NoError(t, err)
+
+			// Confirm that the user is banned
+			assert.Equal(t, toolchainv1alpha1.UserSignupStateLabelValueBanned, userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+			mur, err := hostAwait.GetMasterUserRecord(wait.WithMurName("testuser" + id))
+			require.NoError(t, err)
+			assert.Nil(t, mur)
+			require.NoError(t, err)
+
+			t.Run("verify metrics", func(t *testing.T) {
+				metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 1)
+				metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 0) // not provisioned because banned before signup
+				metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 1)
+				metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // no user on member1
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+			})
+		})
+
+		t.Run("register new user with preexisting ban", func(t *testing.T) {
```

Comment:
the same also in this case - we don't verify metrics logic here, but reg-service